### ISSUE TITLE
Want to mock/stub component method in created/mounted lifecycle. 

### ISSUE THREAD

**Issue by MushengLiaoDavis** [vuejs/vue-test-utils-next#797]
In **vue-test-utils v1.x**, we can mock(stub) component methods for the created/mouted lifecycle as follows:

SFC (class-style component)
```
// ignore other code ...
created() {
}

mounted() {
     this.callThirdPartyLib();
}

callThirdPartyLib() {
    // call third-party libaray
}
```

In test file
```
// use sinon, chai tool

wrapper = shallowMount(MyComponent, {
      methods: {     // In vue-test-utils 2.x  doesn't support mocking methods while mounting component
        callThirdPartyLib: sinon.stub(),     // Don't want to call this real method, so stub it.
        disableTab: sinon.stub(),
        pageScrollTop: sinon.stub()
      },
      localVue,
      store,
});

it('...', () => {
    expect(wrapper.vm.callThirdPartyLib).to.have.been.calledOnce;
})
```

**vue-test-utils v2.x**
I know we can stub the http request action (axios) in created/mounted lifecycle. This behavior is independent (gloabl function or gloabl instance) from component, so  they are easy to stub. But in my scenario, just don't want to call real method (callThirdPartyLib) while mounting component.  

```
import axios from 'axios';

sinon.stub(axios, 'get').resolves({
  data: {
            // some data here
        }
})
```

**Comment by cexbrayat** [comment]
Duplicates of #790 and related to #775

As you are pointing out you can stub calls to 3rd party libs, as you are doing for axios.
Let's say you component looks like:
```ts
import lib from 'lib';
// ...
callThirdPartyLib() {
  lib.something();
}
```

Then your test can mock `lib.something()`. I'm usually using Jest but you get the idea:

```ts
const something = jest.fn();
jest.mock('lib', () => ({
  something
}));

const wrapper = mount(MyComponent);
expect(something).toHaveBeenCalled();
```

Would that be enough for your use-case?

**Comment by MushengLiaoDavis** [comment]
Hi cexbrayat:

      Thanks for your suggestion. I mean vue-test-utils v1.x can support what I want (stub method while mounting component), but v2.x doesn't support. 

For vue3, it focus on the composition-api and function-based programming. Maybe I need to change my old point of view, all logical code are defined in another file and component file is responsible for html layout. As a result, I can easily mock all the methods.

**Comment by cexbrayat** [comment]
Yes, I think we agree 👍 

The internal model of Vue 3 is different, so I don't think we can offer a similar thing to `methods`. Or maybe at the price of hacks that would cost us a lot in maintenance. My recommendation would be to extract your logic in a separate file, and use the pattern I exposed above. I do this with all my Vue 3 projects, and this is working well.

I'm going to close the issue, as we discussed it here and in other related ones to end up on the same conclusion for now.
